src/modules/Threshold/Module.js

```diff
@@ -1,20 +1,26 @@
 /*
  * Image thresholding with 'image-filter-threshold'
  */
-module.exports = function ImageThreshold(options) {
+module.exports = function ImageThreshold(options, UI) {
   options = options || {};
   options.title = "Threshold image";
-  options.threshold = options.threshold || 30;
+  options.threshold = options.threshold || 50;
 
-  var image;
+  UI.onSetup(options.step);
 
-  function draw(inputImage) {
+  var output;
+
+  function draw(input, callback) {
+
+    UI.onDraw(options.step);
+    var image = new Image();
+    image.src = input.src;
     var canvas = document.createElement('canvas');
-    canvas.width = inputImage.naturalWidth;
-    canvas.height = inputImage.naturalHeight;
+    canvas.width = image.naturalWidth;
+    canvas.height = image.naturalHeight;
```

Comment:
This will work only on browsers, right? I think it's pretty easy to write a threshold module - we could as well do it without using the native Image and Canvas classes.